week2.md

```diff
@@ -41,64 +41,105 @@ WORKDIR /fuzz
-Finally, run `make install`.
-This will install Xpdf into the directory we specified earlier.
-Go back to our `xpdf `directory with `cd ..`.
-If you run `ls install/bin` now you should see several executables including `pdftotext`, which is the one that we will fuzz.
+To fuzz effectively, the fuzzer needs a way to monitor the execution of the target program so that it knows which inputs generate interesting new behavior. We will compile the target with a special compiler which inserts instrumentation code that allows Honggfuzz to track code coverage. Honggfuzz can also monitor the target using hardware features on some CPUs or the QEMU emulator.
+
+Run the following command to create a directory for fuzzing Xpdf.
+
+```bash
+mkdir -p work/xpdf
+```
+
+The `-p` flag tells `mkdir` to create the `work` directory if it doesn't already exist. Go into the directory with the following command.
+
+```bash
+cd work/xpdf
+```
+
+Next, download the Xpdf source code using the `curl` utiliy.
+
+```bash
+curl -LO 'https://dl.xpdfreader.com/old/xpdf-3.02.tar.gz'
+```
+
+The `-L` flag tells curl to follow redirects, and the `-O` flag makes curl automatically determine the output file name. The downloaded file is a tar archive compressed using gzip. To extract it, run the following command.
+
+```bash
+tar -xvzf xpdf-3.02.tar.gz
+```
+
+Here, `x` means extract, `v` means verbose (print out file names while extracting), `z` means the file is compressed with gzip, and `f` means read from the file named in the next argument instead of stdin. Move into the newly created directory.
+
+```bash
+cd xpdf-3.02
+```
+
+Now we will build Xpdf using the GNU build system. First, we need to generate a Makefile containing the commands that need to be executed in order to build Xpdf.
+
+```bash
+CC=hfuzz-clang CXX=hfuzz-clang++ ./configure --prefix=/fuzz/work/xpdf/install
+```
+
+`CC=hfuzz-clang CXX=hfuzz-clang++` sets the C and C++ compilers to the Honggfuzz compilers which will instrument the program. The `--prefix` option sets the directory where Xpdf will be installed after it is built. Next, build the program using `Make`. Run `make -j <num_cores>` where `<num_cores>` is the number of CPU cores on your computer, which you can determine by running the `nproc` command. An example is shown below.
+
+```bash
+$ nproc
+4
+$ make -j 4
+```
+
+This will compile Xpdf by executing the commands in the Makefile. The `-j` option tells Make how many jobs to run in parallel. Finally, install Xpdf into the directory we specified earlier using the following command.
+
+```bash
+make install
+```
+
+Go back to our `xpdf`directory.
+
+```bash
+cd ..
+```
+
+If you run `ls install/bin` now you should see several executables including `pdftotext`, which is the one that we will fuzz. An example is shown below of what the output should look like.
+
+```bash
+[root@801c9602156c xpdf]# ls install/bin/
+pdffonts  pdfimages  pdfinfo  pdftops  pdftotext
+```
```

Comment:
Code blocks shouldn't be marked as bash if they include the shell prompt or command output.